src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -187,14 +187,15 @@ public static class TimeSeriesCatalog
         /// It is used when score is calculated for each root cause item. The range of beta should be in [0,1].
         /// For a larger beta, root cause items which have a large difference between value and expected value will get a high score.
         /// For a small beta, root cause items which have a high relative change will get a low score.</param>
+        /// <param name="rootCauseThreshold">A threshold to determine whether the point should be root cause. If the point's delta is equal to or larger than anomalyDeltaThreshold multiplies anomaly dimension point's delta, this point is treated as a root cause. Different threshold will turn out different result. Users can choose the delta according to their data and requirment. </param>
         /// <example>
         /// <format type="text/markdown">
         /// <![CDATA[
         /// [!code-csharp[LocalizeRootCause](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/LocalizeRootCause.cs)]
         /// ]]>
         /// </format>
         /// </example>
-        public static RootCause LocalizeRootCause(this AnomalyDetectionCatalog catalog, RootCauseLocalizationInput src, double beta = 0.5)
+        public static RootCause LocalizeRootCause(this AnomalyDetectionCatalog catalog, RootCauseLocalizationInput src, double beta = 0.5, double rootCauseThreshold = 0.95)
```

Comment:
> nit: another option as suggested by Harish (for AD) is to create [an Options class](https://github.com/dotnet/machinelearning/blob/c0eeea92533eac507a0958a170bce4a270a7f9d2/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L1537) to avoid multiple overload in future.

I found that the option is always served for transformer implementation. Does it makes sense to use it in the extentions catalog?

Reply:
I think ML.Net team treats it at a pattern of handling multiple overloaded methods thus not confined to which type of class it can be used. yet it's just a minor comments so feel free to ignore

Reply:
Four parameters to a function is not too bad. If you are expecting to enhance this API further in the future by adding more parameters, I would suggest adding an Options parameter now. If there are no plans to add new parameters now, please add an Options parameter the next time there is a new parameter added.

---
In reply to: [438493061](https://github.com/dotnet/machinelearning/pull/5218#discussion_r438493061) [](ancestors = 438493061)

Reply:
> Four parameters to a function is not too bad. If you are expecting to enhance this API further in the future by adding more parameters, I would suggest adding an Options parameter now. If there are no plans to add new parameters now, please add an Options parameter the next time there is a new parameter added.
> 
> E.g. We have now migrated rootCauseThreshold from being a private constant in RootCauseAnaylyzer to being a parameter here. But I also see _anomalyRatioThreshold and _anomalyPreDeltaThreshold as private constants there?
> 
> Does it make sense to convert them to parameters? If so, please create an Options class and add all those parameters there.
> 
> In reply to: [438493061](https://github.com/dotnet/machinelearning/pull/5218#discussion_r438493061) [](ancestors = 438493061)

Thanks for the suggestion, have updated

Reply:
Sorry, I don't think I communicated well. 

The pattern in ML.NET is to have two overloaded functions. A convenience function and a full function. The convenience function takes in a minimal set of parameters that will meet the common use case and use default values for all other parameters. We always have this function. The second function is if a particular feature requires a function with lots of parameters. In this case we add a second overloaded function that takes an Options parameter where the user can set each parameter individually. 

So at the very least you should have one function *without* an Options parameter. You add a second function with an Options parameter only if you are going to need lots of parameters. 

Like I mentioned above, if all you need are four parameters, a single function is okay. Only if you were going to migrate the other constants from RootCauseAnalyzer, please add a second function with an Options parameter. 

---
In reply to: [442571173](https://github.com/dotnet/machinelearning/pull/5218#discussion_r442571173) [](ancestors = 442571173)